### PR TITLE
Reconcile CXF bom with current artifacts

### PR DESCRIPTION
**Pull request by cunningt**
Noticed there were some missing BOM entries for some of the CXF artifacts, and could not find cxf-services-sts-war.

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -156,12 +161,12 @@
             </dependency>
             <dependency>
                 <groupId>org.apache.cxf</groupId>
-                <artifactId>cxf-rt-management</artifactId>
+                <artifactId>cxf-rt-javascript-tests</artifactId>
```
Comment (reta) [inline bom/pom.xml:164]: The tests artifacts (like `cxf-rt-javascript-tests`) should very likely not be there

File: bom/pom.xml [hunk 2]
```diff
@@ -264,6 +284,26 @@
                 <artifactId>cxf-rt-rs-service-description</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.cxf</groupId>
+                <artifactId>cxf-rt-rs-service-description-openapi-v3</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+                        <dependency>
```
Comment (reta) [inline bom/pom.xml:292]: The `cxf-rt-rs-service-description-common-openapi` should probably not be there as well, it is more or less internal to CXF implementation

File: bom/pom.xml [hunk 3]
```diff
@@ -384,6 +429,11 @@
                 <artifactId>cxf-rt-security</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (reta) [inline bom/pom.xml:432]: This is one is certainly should not be there, this is internal OSGi bundle and should not be consumed by applications or services directly.

File: bom/pom.xml [hunk 4]
```diff
@@ -449,6 +504,11 @@
                 <artifactId>cxf-tools-wsdlto-frontend-jaxws</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (reta) [inline bom/pom.xml:507]: Also, test one, please remove

File: bom/pom.xml [hunk 5]
```diff
@@ -459,6 +519,11 @@
                 <artifactId>cxf-codegen-plugin</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (reta) [inline bom/pom.xml:522]: AFAIK plugins are not imported as part of BOMs (please correct me if I am wrong), should not be there

File: bom/pom.xml [hunk 6]
```diff
@@ -566,6 +626,11 @@
                 <artifactId>cxf-services-xkms-features</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (reta) [inline bom/pom.xml:629]: Also test one, please remove

**Comment by reta** [comment]
@cunningt thanks a lot for taking a lead on this one (always wanted to do that), a few minor comments, but looks good otherwise

File: bom/pom.xml (hunk 3 above)
Comment (cunningt) [reply bom/pom.xml:412]: Removed cxf-servlet-compatible

File: bom/pom.xml (hunk 4 above)
Comment (cunningt) [reply bom/pom.xml:487]: Removed cxf-tools-wsdlto-test

File: bom/pom.xml (hunk 5 above)
Comment (cunningt) [reply bom/pom.xml]: Removed maven-plugins

File: bom/pom.xml (hunk 6 above)
Comment (cunningt) [reply bom/pom.xml]: cxf-services-xkms-itests

File: bom/pom.xml (hunk 1 above)
Comment (cunningt) [reply bom/pom.xml]: Removed cxf-rt-javascript-tests

File: bom/pom.xml (hunk 2 above)
Comment (cunningt) [reply bom/pom.xml]: Removed cxf-rt-rs-service-description-common-openapi

**Comment by cunningt** [comment]
Removed the test artifacts / internal artifacts and removed the maven plugins.

File: bom/pom.xml [hunk 7]
```diff
@@ -264,6 +279,16 @@
                 <artifactId>cxf-rt-rs-service-description</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (reta) [inline bom/pom.xml:282]: Apologies, I missed this one, the `cxf-rt-rs-service-description-swagger` is gone in `main` (the 1.x does not support Jakarta), could you please remove it? (I will bring it back on backports). 

[1] https://github.com/apache/cxf/pull/1000

Comment (cunningt) [reply bom/pom.xml:282]: No worries, removed!

**Review by reta — APPROVED** [review]

**Pull request merged**